File: src/main/java/frc/utils/controlWrappers/ElevatorFF.java
package frc.utils.controlWrappers;

import edu.wpi.first.math.controller.ElevatorFeedforward;

/*
 * wrapper for ElevatorFeedforward that allows setting gains from a record
 */
public class ElevatorFF extends ElevatorFeedforward{
    public ElevatorFF(double kS, double kG, double kV, double kA){
        super(kS, kG, kV, kA);
    }
    public ElevatorFF(double kS, double kG, double kV, double kA, double dt){
        super(kS, kG, kV, kA, dt);
    }
    public ElevatorFF(PIDGains.GravityFF gains){
        super(gains.kS(), gains.kG(), gains.kV(), gains.kA());
    }
}

File: src/main/kotlin/frc/utils/controlWrappers/ElevatorFF.kt
package frc.utils.controlWrappers

import edu.wpi.first.math.controller.ElevatorFeedforward

/*
 * wrapper for ElevatorFeedforward that allows setting gains from a record
 */
class ElevatorFF : ElevatorFeedforward {
    constructor(kS: Double, kG: Double, kV: Double, kA: Double) : super(kS, kG, kV, kA)
    constructor(kS: Double, kG: Double, kV: Double, kA: Double, dt: Double) : super(kS, kG, kV, kA, dt)
    constructor(gains: PIDGains.GravityFF) : super(gains.kS, gains.kG, gains.kV, gains.kA)
}
